samples/kafka_confluent/sender/main.go

```diff
@@ -43,11 +83,16 @@ func main() {
 			"message": "Hello, World!",
 		})
 
-		if result := c.Send(
-			// Set the producer message key
-			confluent.WithMessageKey(ctx, e.ID()),
-			e,
-		); cloudevents.IsUndelivered(result) {
+		result := c.Send(confluent.WithMessageKey(ctx, e.ID()), e)
+		if result != nil {
+			if val, ok := result.(kafka.Error); ok && val.Code() == kafka.ErrQueueFull {
+				// Producer queue is full, wait 1s for messages
+				// to be delivered then try again.
+				time.Sleep(time.Second)
```

Comment:
Is this a best practice? Wondering if it complicates the example, unless you think it's always important to implement this logic.

Reply:
I've seen them use assert like the above when sending messages asynchronously.

But I tried to set the producer buffer channel size with 1 `"go.produce.channel.size": 1`, and sending 10,000 messages within 1 second several times. It never catches the `kafka.ErrQueueFull` error. So let's remove it.